cli/loader/src/lib.rs

```diff
@@ -521,11 +521,18 @@ impl Loader {
             .and_then(|n| n.to_str())
             .and_then(|file_name| self.language_configuration_ids_by_file_type.get(file_name))
             .or_else(|| {
-                path.extension()
-                    .and_then(|extension| extension.to_str())
-                    .and_then(|extension| {
-                        self.language_configuration_ids_by_file_type.get(extension)
-                    })
+                let mut path = path.to_owned();
+                let mut full_extension = String::new();
+                while let Some(extension) = path.extension() {
+                    full_extension = format!(
+                        "{}{}{full_extension}",
+                        extension.to_str()?,
+                        if full_extension.is_empty() { "" } else { "." }
+                    );
+                    path = PathBuf::from(path.file_stem()?.to_os_string());
+                }
+                self.language_configuration_ids_by_file_type
+                    .get(&full_extension)
```

Comment:
for clarity and avoiding allocations on every call to format, store a `Vec` of extensions, and push to it on each iteration while there is an extension, then just join them with a `.` at the end